### ISSUE TITLE
Consume elemental-cli install

### ISSUE THREAD

**Issue by mudler** [rancher-sandbox/cOS-toolkit#1072]
Consume elemental-cli in place of cos-installer in cos-toolkit:

- [x] modify system/cos so that uses elemental-cli for installation #1074
- [x] make sure it passes tests and maybe validate some of the scenarios that are not covered manually too https://github.com/rancher-sandbox/cOS-toolkit/issues/581
- [x] check harvester [impact](https://github.com/harvester/harvester-installer/blob/3f1dbfccacc7c2aa57ec84449d27719d2586b59f/package/harvester-os/files/usr/sbin/harv-install#L306)

**Comment by davidcassany** [comment]
Starting with this one, JFYI  I will create a very early draft mostly to run CI.

**Comment by davidcassany** [comment]
@bk201 I hacked around harvester installer to check the impact of the installer code change.  I modified the rootfs image creation from this [Dockerfile](https://github.com/harvester/harvester-installer/blob/3f1dbfccacc7c2aa57ec84449d27719d2586b59f/package/harvester-os/Dockerfile) and added a COPY statement to include the `elemental` binary and I also modified the install [script](https://github.com/harvester/harvester-installer/blob/3f1dbfccacc7c2aa57ec84449d27719d2586b59f/package/harvester-os/files/usr/sbin/harv-install) to make use of the copied `elemental binary`.

With the following change the installation succeeded:

```bash
_COS_BOOTING_FROM_LIVE="true" _COS_INSTALL_ISO_URL="" INTERACTIVE=yes _COS_INSTALL_DEVICE="$install_device" cos install
```

to

```bash
elemental install "${install_device}"
```

`_COS_INSTALL_ISO_URL=""` is already the default and the same for `INTERACTIVE=yes`. I am not sure why you needed `_COS_BOOTING_FROM_LIVE="true"`, but within the new code there is not such a config.

Given that harvester does not make use of the `cos-installer` and calls `cos install` instead I believe that merging #1074 will not have any effect as it basically sets `cos-installer` to make use of `elemental install`, but `cos.sh` will be still provided after merging, so `cos install` call will remain unchanged.
